### PR TITLE
Initial tests to verify a new connection

### PR DESCRIPTION
**Pull request by Jeconias**
image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,5 +25,10 @@
     "ejs": "^2.5.5",
     "express": "^4.14.0",
     "socket.io": "^1.7.2"
+  },
+  "devDependencies": {
+    "chai": "^4.2.0",
+    "mocha": "^8.2.0",
+    "socket.io-client": "^2.3.1"
```
Comment (LalitNM) [inline package.json:32]: Do not update the dependencies, because this crashed the app on heroku.

File: package.json [hunk 2]
```diff
@@ -4,11 +4,13 @@
...
   "scripts": {
-    "start": "node app.js"
+    "start": "node app.js",
+    "test:watch": "mocha --watch ./*.test.js",
+    "test": "mocha ./*.test.js"
   },
   "engines": {
     "npm": "3.10.10",
-    "node": "6.9.4"
+    "node": "12.19.0"
```
Comment (LalitNM) [inline package.json:13]: Do not update the dependencies, because this crashed the app on heroku.

File: app.js [hunk 3]
```diff
@@ -28,12 +28,11 @@ io.on('connection', (socket) => {
...
 	//When client requests for setting username
 	socket.on('set username', (name) => {
-		name = name.trim()
+		name = (name || "").trim()
 
 		//if name is empty(null), do nothing
-		if(name == "" || name == null) {
-			return;
-		}
+		if(!name) return socket.emit('user invalid', {username: `${name} is invalid.`});
```
Comment (LalitNM) [inline app.js]: If you are adding this here then you have to add this socket in `client.js` too.

File: app.js [hunk 4]
```diff
@@ -28,12 +28,11 @@ io.on('connection', (socket) => {
 
 	//When client requests for setting username
 	socket.on('set username', (name) => {
-		name = name.trim()
+		name = (name || "").trim()
```
Comment (LalitNM) [inline app.js:32]: Any specific reason to change this?

**Review by LalitNM — CHANGES_REQUESTED** [review]
Some additional changes

- Shift all test files in another directory named `test`

- Remove `yarn.lock` from your commits, this is just installation files.

File: package.json (hunk 2 above)
Comment (Jeconias) [reply package.json:13]: mocha@8.2.0 expect a version of node >= 10.12.0. Any suggestion?

File: app.js (hunk 4 above)
Comment (Jeconias) [reply app.js:32]: If the socket.io client not send a name (undefined), its code will break.

File: app.js (hunk 3 above)
Comment (Jeconias) [reply app.js]: Sorry, I forgot.

File: package.json (hunk 2 above)
Comment (LalitNM) [reply package.json:13]: Okay, leave it, I just tested this version of app. It is working fine.

File: app.js (hunk 3 above)
Comment (Jeconias) [reply app.js]: Added the new event to the client. 4dc36b6

**Comment by Jeconias** [comment]
@LalitNM, I defined the version mocha in 6.2.3 to enable the use with node 6.9.4.

File: package.json [hunk 5]
```diff
@@ -27,8 +27,8 @@
     "socket.io": "^1.7.2"
   },
   "devDependencies": {
-    "chai": "^4.2.0",
-    "mocha": "^8.2.0",
-    "socket.io-client": "^2.3.1"
+    "chai": "4.2.0",
+    "mocha": "6.2.3",
+    "socket.io-client": "2.3.1"
```
Comment (LalitNM) [inline package.json]: Can you just revert this back to previous version. This is acceptable on heroku.

File: package.json [hunk 6]
```diff
@@ -5,12 +5,12 @@
...
     "start": "node app.js",
-    "test:watch": "mocha --watch ./*.test.js",
-    "test": "mocha ./*.test.js"
+    "test:watch": "mocha --watch ./tests/**/*.test.js",
+    "test": "mocha ./tests/**/*.test.js"
   },
   "engines": {
     "npm": "3.10.10",
-    "node": "12.19.0"
+    "node": "6.9.4"
```
Comment (LalitNM) [inline package.json]: Also revert it to `12.19.0`.

File: package.json (hunk 5 above)
Comment (Jeconias) [reply package.json]: Done!

File: package.json (hunk 6 above)
Comment (Jeconias) [reply package.json]: Done!

**Comment by LalitNM** [comment]
@Jeconias these many changes are enough for this pull request. I will look into the versions of all dependencies and will rectify the errors. Then I will finally merge this pull. Work from your end is done for this pull request.

Meanwhile can you list some more tests in [this](https://gitter.im/Baat-Cheet-IIT/Unit-tests). This will help in opening more issue on tests.

**Review by LalitNM — APPROVED** [review]

**Pull request merged**